dev/com.ibm.ws.config/src/com/ibm/ws/config/xml/internal/ConfigRefresher.java

```diff
@@ -265,4 +282,30 @@ public void variableRefresh(Map<String, DeltaType> deltaMap) {
         doRefresh(deltaMap);
 
     }
+
+    // Register as a multi-threaded hook to invoke changesEnded() in
+    // multi-threaded mode when restoring from checkpoint. Invoking
+    // changesEnded() in single-threaded mode may fail restore as it
+    // performs a blocking operation.
+    private class ChangesEndedHook implements CheckpointHook {
+
+        private boolean runMultiThreaded = false;
+
+        // Set before restore is called
+        private RuntimeUpdateNotification configUpdatesDelivered = null;
+
+        @Override
+        public void prepare() {
+            runMultiThreaded = true;
+        }
+
+        @Override
+        public void restore() {
+            if (runMultiThreaded) {
+                changesEnded(configUpdatesDelivered);
```

Comment:
set `runMultiThreaded` to false here so if other config updates happen after restore they will work.

Reply:
Done